hexrd/rotations.py

```diff
@@ -536,159 +516,52 @@ def expMapOfQuat(quats):
     cdim = 4  # critical dimension of input
     quats = np.atleast_2d(quats)
     if len(quats) == 1:
-        assert quats.shape[1] == cdim, \
+        assert quats.shape[1] == cdim, (
             "your input quaternion must have %d elements" % cdim
+        )
         quats = np.reshape(quats, (cdim, 1))
     else:
-        assert len(quats) == cdim, \
+        assert len(quats) == cdim, (
             "your input quaternions must have shape (%d, n) for n > 1" % cdim
+        )
 
-    # ok, we have hstacked quats; get angle
-    phis = 2.*arccosSafe(quats[0, :])
-
-    # now axis
-    ns = unitVector(quats[1:, :])
-
-    # reassemble
-    expmaps = phis*ns
-    return expmaps.squeeze()
+    return _quat_to_scipy_rotation(quats).as_rotvec().T.squeeze()
 
 
-def rotMatOfExpMap_opt(expMap):
-    """Optimized version of rotMatOfExpMap
+def rotMatOfExpMap(expMap):
+    """
+    Make a rotation matrix from an expmap
     """
     if expMap.ndim == 1:
         expMap = expMap.reshape(3, 1)
 
-    # angles of rotation from exponential maps
-    phi = atleast_1d(columnNorm(expMap))
-
-    # skew matrices of exponential maps
-    W = skewMatrixOfVector(expMap)
-
-    # Find tiny angles to avoid divide-by-zero and apply limits in expressions
-    zeroIndex = phi < cnst.epsf
-    phi[zeroIndex] = 1
-
-    # first term
-    C1 = sin(phi) / phi
-    C1[zeroIndex] = 1  # is this right?  might be OK since C1 multiplies W
-
-    # second term
-    C2 = (1 - cos(phi)) / phi**2
-    C2[zeroIndex] = 0.5  # won't matter because W^2 is small
-
-    numObjs = expMap.shape[1]
-    if numObjs == 1:  # case of single point
-        W = np.reshape(W, [1, 3, 3])
-
-    C1 = np.tile(
-        np.reshape(C1, [numObjs, 1]),
-        [1, 9]).reshape([numObjs, 3, 3])
-    C2 = np.tile(
-        np.reshape(C2, [numObjs, 1]),
-        [1, 9]).reshape([numObjs, 3, 3])
-
-    W2 = np.zeros([numObjs, 3, 3])
-
-    for i in range(3):
-        for j in range(3):
-            W2[:, i, j] = np.sum(W[:, i, :]*W[:, :, j], 1)
-
-    rmat = C1*W + C2 * W2
-    rmat[:, 0, 0] += 1.
-    rmat[:, 1, 1] += 1.
-    rmat[:, 2, 2] += 1.
-
-    return rmat.squeeze()
-
-
-def rotMatOfExpMap_orig(expMap):
```

Comment:
We can't technically be sure people aren't using this (why would they?) so let's deprecate this using the previous strategy instead of deleting it. We can remove the body of the function, and leave the signature with the deprecated decorator.

Reply:
Fixed